### PR TITLE
Replace std::strcmp by std::string_view::operator==

### PR DESCRIPTION
**Pull request by litelawliet**
Thanks a lot for making a contribution to SFML! 🙂

Before you create the pull request, we ask you to check the follow boxes. (For small changes not everything needs to ticked, but the more the better!)

* [x] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before?
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description

This PR is related to the issue https://github.com/SFML/SFML/issues/2366

I couldn't find all the references found in the issue but it should be exhaustive for master branch. Feel free to point out any missing references or any remarks.

## Tasks

* [ ] Tested on Linux
* [x] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
Why couldn’t you find those other strcmp uses?

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
I double checked and `master` does include more uses of `std::strcmp` to remove so please fix those in this PR too.

**Comment by litelawliet** [comment]
Thank you for the list, somehow VS couldn't find more references in the solution, I reconfigured the project via CMake and that should be all now

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: examples/vulkan/Vulkan.cpp [hunk 1]
```diff
@@ -470,11 +470,11 @@ class VulkanExample
             // -- VK_LAYER_GOOGLE_unique_objects
             // These layers perform error checking and warn about bad or sub-optimal Vulkan API usage
             // VK_LAYER_LUNARG_monitor appends an FPS counter to the window title
-            if (!std::strcmp(layer.layerName, "VK_LAYER_LUNARG_standard_validation"))
+            if (std::string_view(layer.layerName) == std::string_view("VK_LAYER_LUNARG_standard_validation"))
```
Comment (ChrisThrasher) [inline examples/vulkan/Vulkan.cpp:473]:
```suggestion
            if (std::string_view(layer.layerName) == "VK_LAYER_LUNARG_standard_validation")
```
We can compare string views to `const char*`s whether they're Vulkan constant or string literals so we don't need to explicitly construct a second string view. This applies to many other places in this PR so I'll only mention it once.

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]

File: examples/vulkan/Vulkan.cpp (hunk 1 above)
Comment (litelawliet) [reply examples/vulkan/Vulkan.cpp]: You're right, I completely overlook string literals in the process. Commit 2767be2e9e7961f7a4bc6032be9d957f6ed6c20c should fix those instances.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
Please build the `format` target which will run clang-format and then those formatting CI jobs will start passing.

**Comment by litelawliet** [comment]
Would it be possible to support LLVM 15 and above ? I just saw the project only support up to LLVM 14

**Comment by ChrisThrasher** [comment]
No because clang-format-15 produces different formatted code than versions 12 through 14.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by ChrisThrasher — APPROVED** [review]

File: src/SFML/Window/DRM/DRMContext.cpp [hunk 2]
```diff
@@ -365,7 +365,7 @@ int initDrm(sf::priv::Drm& drm, const char* device, const char* modeStr, unsigne
         {
             drmModeModeInfoPtr currentMode = &connector->modes[i];
 
-            if (std::strcmp(currentMode->name, modeStr) == 0)
+            if (std::string_view(currentMode->name) == std::string_view(modeStr))
```
[the remark this answers is not in the document]
Comment (litelawliet) [reply src/SFML/Window/DRM/DRMContext.cpp]: Right, fixed in d124e7164f9185f2550f8c03b4ec8b02dd2ca30e

**Review by kimci86 — APPROVED** [review]

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
Can we remove `#include <cstring>` from any of these files now?

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by litelawliet** [comment]
So after a few builds (sorry for that) that I couldn't test for Linux GCC and Clang, it appears the only
`#include <cstring>` we can remove are in `SFML\Window\Unix\VulkanImplX11.cpp` and `SFML\Window\Win32\VulkanImplWin32.cpp` files.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
For whatever reason the CI pipeline isn't running? At least it appears so to me. Can you run `git commit --amend` then `git push --force` to change the commit hash and push that so CI gets triggered again? Unsure why it's acting up.

**Comment by litelawliet** [comment]
Just did the two commands you suggested, doesn't look like CI is running ?

I generally interactively rebase my branch, choose to squash, fixup or edit commits I did (usually to have only one remaining) then force push the entire thing, could it introduce any problem for CI ?

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
> I generally interactively rebase my branch, choose to squash, fixup or edit commits I did (usually to have only one remaining) then force push the entire thing, could it introduce any problem for CI ?

This is perfectly fine. I think we just hit a weird one-time issue.

**Review by eXpl0it3r — APPROVED** [review]

**Pull request merged**